src/UI/Art/SpriteArtist/SpriteMapView.java

```diff
@@ -258,14 +271,28 @@ public void render(Graphics g)
       // Update the central sprite indices so animations happen in sync.
       updateAnimationIndices();
 
+      // Here are the fog-drawing rules. If there are:
+      //   zero humans - spectating - draw everything the current player sees.
+      //   one human - player vs ai - draw everything the human player could see.
+      //   2+ humans - player vs player - draw what the current player sees, IFF the player is human.
+      GameMap gameMap = myGame.activeCO.myView; // Start by assuming zero humans.
+      int numHumans = countHumanPlayers(myGame);
+      if( 1 == numHumans )
+      {
+        // Since there is only one human, always use the human's vision to determine what is drawn.
+        gameMap = getHumanPlayerMap(myGame);
+      }
+      // Don't let humans peek in on what the AI is doing.
+      boolean drawFogEverywhere = myGame.isFogEnabled() && (numHumans > 1) && (myGame.activeCO.isAI());
```

Comment:
Ehh, I'm not sure I like fake COs, and it's been an explicit design decision to keep the game-logic code unaware of the UI/drawing code - it should be up to the UI side to pull whatever data it needs from the logic side.
On the other hand, I also don't like having that logic in two places; the normal fix is to just push it into a function, but I needed both `drawFogEverywhere` and `gameMap` to come out of that function, so that wasn't ideal. I guess just take a stab at it if you have an idea.